### PR TITLE
feat(meta): add lease for meta and remove check_exists in hummock

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

## What's changed and what's your intention?

close https://github.com/singularity-data/risingwave/issues/2515
In this PR, we only need to check whether the leader info changed when meta-service may reload after panic, or be partitioned with etcd.
If some other node became new meta-node, it will changed leader info and all transaction submitted by the origin meta-node will fail.

## Checklist

- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/meta/src/hummock/hummock_manager.rs [hunk 1]
```diff
@@ -269,17 +271,26 @@ where
...
                 // The worker is not found in cluster.
                 return Err(Error::InvalidContext(context_id));
             }
         }
+        trx.check_equal(
+            META_CF_NAME.to_owned(),
+            META_LEADER_KEY.as_bytes().to_vec(),
+            info.encode_to_vec(),
+        );
         meta_store.txn(trx).await.map_err(Into::into)
```
Comment (zwang28) [inline src/meta/src/hummock/hummock_manager.rs:294]: What if context_id becomes invalid just before `meta_store.txn`?

Comment (Little-Wallace) [reply src/meta/src/hummock/hummock_manager.rs:294]: Good catch.... It seems that I need a memory lock for `context_id`, does it?

Comment (zwang28) [reply src/meta/src/hummock/hummock_manager.rs:294]: > Good catch.... It seems that I need a memory lock for `context_id`, does it?

Yeah something prevents `ClusterManager` from `delete_worker_node`.

**Comment by zwang28** [comment]
LGTM. 
@yezizp2012 FYI we'd like to add meta lease.

File: src/meta/src/lib.rs [hunk 2]
```diff
@@ -65,7 +67,10 @@ enum Backend {
 pub struct MetaNodeOpts {
     // TODO: rename to listen_address and separate out the port.
     #[clap(long, default_value = "127.0.0.1:5690")]
-    host: String,
+    listen_addr: String,
```
Comment (yezizp2012) [inline src/meta/src/lib.rs:70]: Plz fix the opt in risedev as also. https://github.com/singularity-data/risingwave/blob/ce23e4a099867e10f2426eefc00d7475175b1e33/src/risedevtool/src/task/meta_node_service.rs#L45

**Comment by yezizp2012** [comment]
LGTM.

File: src/meta/src/lib.rs (hunk 2 above)
Comment (Little-Wallace) [reply src/meta/src/lib.rs:70]: ok.

**Comment by Little-Wallace** [comment]
I talked about lease and context id with @zwang28 .
When some node lost connection with meta, the timestamp of it would not be updated and meta-service will remove this node and call `release_context`. But the operation `release_context` and `report_compact_task` or other operations in hummock manager are all blocked by `RwLock`. So if `ClusterManager` removed `context_id` after some thread check context_id and just before it write to meta-storage,  it would be blocked by compaction until other thread finished writing to meta-storage. And then, it can see the results of the previous operations.

**Comment by skyzh** [comment]
This PR breaks the usage of `--hosts` args. Next time please inform us to update docker and risingwave-operator args, so that everything will work fine.

cc @Nebulazhang @Little-Wallace

**Pull request merged**